### PR TITLE
password recovery w/tests

### PR DESCRIPTION
**Pull request by t-romani**

### CODE REVIEW

File: app/controllers/api/v1/api_controller.rb [hunk 1]
```diff
@@ -18,9 +18,9 @@ def render_error_response(error)
       end
 
       def render_not_found_response
-        render json: { error: I18n.t(
-          'api.error.invalid_request.content_not_found'
-        ) }, status: :not_found
+        render json: { error: I18n
+          .t('api.error.invalid_request.content_not_found') },
+               status: :not_found
```
Comment (gonzaM) [inline app/controllers/api/v1/api_controller.rb:24]: Why this change?

File: app/views/api/v1/passwords/update.json.jbuilder [hunk 2]
```diff
@@ -0,0 +1,3 @@
+json.success true
+json.user user, partial: 'api/v1/users/info', as: :user
+json.message 'Your password has been successfully updated.'
```
Comment (gonzaM) [inline app/views/api/v1/passwords/update.json.jbuilder]: I think this and the `success` key are not necessary.
You're rendering this view because was a success, the client receives 200 so it's not necessary to explicitly mention the "success true". Same for the message. The client can show a message if they want.

File: app/views/api/v1/users/_info.json.jbuilder [hunk 3]
```diff
@@ -1,3 +1,3 @@
 json.id       user.id
 json.name     user.full_name
-json.gender   user.gender
+json.email    user.email
```
Comment (gonzaM) [inline app/views/api/v1/users/_info.json.jbuilder]: Missing eof
(You can configure your editor to add an end of file always)

File: db/migrate/20190807183229_add_devise_to_users.rb [hunk 4]
```diff
@@ -10,6 +10,7 @@ def change
       ## Recoverable
       t.string   :reset_password_token
       t.datetime :reset_password_sent_at
+      t.boolean  :allow_password_change, default: false, null: false
```
Comment (gonzaM) [inline db/migrate/20190807183229_add_devise_to_users.rb]: You should not modify existing migrations. If you want to add something, add a new one.
Let me know if you don't know why.

**Review by gonzaM — COMMENTED** [review]
Good job. Small comments

File: app/controllers/api/v1/api_controller.rb (hunk 1 above)
Comment (t-romani) [reply app/controllers/api/v1/api_controller.rb:24]: not sure, probably trying to make it look prettier 🤔

Comment (gonzaM) [reply app/controllers/api/v1/api_controller.rb:24]: haha ok, I think it is a weird indentation, but it's up to you 😄

File: spec/request/password/update_spec.rb [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+    it 'returns correct user' do
+      subject
+      expect(parsed_data['user']['id']).to eq(user.id)
+    end
+  end
+
+  context 'when invalid' do
+    let!(:auth_headers) { {} }
+
+    it 'return unauthorized status' do
```
Comment (gonzaM) [inline spec/request/password/update_spec.rb]: returns

**Review by gonzaM — APPROVED** [review]

File: app/controllers/api/v1/api_controller.rb (hunk 1 above)
Comment (t-romani) [reply app/controllers/api/v1/api_controller.rb:24]: fair enough, what would not be weird @gonzaM ?
Anyway ill probably be weird until i finally make friends with rubo🤖

File: app/views/api/v1/users/_info.json.jbuilder (hunk 3 above)
Comment (JuanfraM) [inline app/views/api/v1/users/_info.json.jbuilder:3]: why was removed gender?

File: spec/request/password/create_spec.rb [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+
+  subject do
+    post user_password_path, params: {
+      email: user[:email]
+    }, as: :json
+  end
+
+  it 'returns ok status' do
+    subject
+    expect(response).to have_http_status(:ok)
```
Comment (JuanfraM) [inline spec/request/password/create_spec.rb]: `be_success`

**Review by JuanfraM — APPROVED** [review]
Nice job!
Minor comments

File: app/views/api/v1/users/_info.json.jbuilder (hunk 3 above)
Comment (t-romani) [reply app/views/api/v1/users/_info.json.jbuilder:3]: i thought it was not particularly important

File: spec/request/password/update_spec.rb [hunk 7]
```diff
@@ -0,0 +1,40 @@
...
+    put user_password_path, params: {
+      password: new_password,
+      password_confirmation: new_password
+    }, headers: auth_headers, as: :json
+  end
+
+  context 'when valid' do
+    it 'returns ok status' do
+      subject
+      expect(response).to have_http_status(:ok)
```
Comment (JuanfraM) [inline spec/request/password/update_spec.rb]: `be_successful `
